src/test/compile-fail/rfc-1937-termination-trait/termination-trait-main-i32.rs

```diff
@@ -8,6 +8,7 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-fn main() -> i32 { //~ ERROR main function has wrong type [E0580]
+// error-pattern:`main` can only return types that implement std::process::Termination, not `i32`
```

Comment:
It would be nice if this suggested some types that do work, like `()` and `!`.  (Especially since `Termination` is unstable, which makes these errors more confusing, based on reaction to `Carrier` errors.)

Reply:
I think that's a good idea. In particular, `Result<(), E> where E: Debug` helps me understand the motivation better. Is there any guidance for how long error messages can be?

Reply:
I feel like fitting the `Result` suggestion into an error message is going to make this note too long no matter what, though I'm open to suggestions. So we are left with `()` and `!`.

Only question is, will a new Rust user from C who writes `fn main() -> i32` have any idea what `!` is, and if not, how would they Google it? `()` and `!` are both somewhat cryptic type names, and it seems like a bad experience for someone to come across error message with both of these types potentially very early on in their Rust language usage. That said, I can see how this would be helpful for people from other perspectives, and I'm sure I'm rehashing some argument from the original RFC. Perhaps we can use `never`.

For now, I've only included `()` as an example, with the expectation that someone who really wants to know what's going on can google `std::process::Termination` and read about other types that implement it. (I tried it, the first result gives all implementations!) I'd like to know what other people think about this as well.